javagen/src/main/java/com/azure/autorest/mapper/ServiceClientMapper.java

```diff
@@ -109,14 +109,16 @@ public ServiceClient map(CodeModel codeModel) {
                 serviceClientPropertyClientType = serviceClientPropertyClientType.asNullable();
             }
 
-            boolean serviceClientPropertyIsReadOnly = p.getSchema() instanceof ConstantSchema;
+            // boolean serviceClientPropertyIsReadOnly = p.getSchema() instanceof ConstantSchema;
```

Comment:
Maybe you can set it to false only for `!setting.isFluent()`?

Mgmt won't want user to set the `api-verison`, they are not going to be compatible.

Reply:
Done.